### PR TITLE
Accept arbitrary version formats in overrides.

### PR DESCRIPTION
**Pull request by BillyONeal**
We accidentally implemented the overrides deserializer to reject non-relaxed-versions in "overrides", despite the original intent that overrides are scheme-less. Note how the original schema:

https://github.com/microsoft/vcpkg-tool/pull/1209

, and original documentation:

https://github.com/microsoft/vcpkg-docs/pull/159

said everything should be accepted here.

Also fix our example printing when we're asking users to add overrides to ensure it is valid JSON.

### CODE REVIEW

File: src/vcpkg/sourceparagraph.cpp [hunk 1]
```diff
@@ -150,7 +165,7 @@ namespace vcpkg
         static constexpr StringLiteral DESCRIPTION = "Description";
         static constexpr StringLiteral FEATURE = "Feature";
         static constexpr StringLiteral MAINTAINERS = "Maintainer";
-        static constexpr StringLiteral NAME = "Source";
```
Comment (BillyONeal) [inline src/vcpkg/sourceparagraph.cpp:153]: This took me an embarrassing amount of time to realize that NAME was not always "name".

File: docs/vcpkg-schema-definitions.schema.json [hunk 2]
```diff
@@ -763,26 +763,27 @@
...
               "pattern": "^\\d{4}-\\d{2}-\\d{2}(\\.\\d+)*(#\\d+)?$"
             },
             "version-semver": {
-              "description": "A semantic version string. See https://semver.org/",
+              "description": "Deprecated in favor of 'version' in overrides. A semantic version string. See https://semver.org/",
               "type": "string",
               "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?(#\\d+)?$"
             },
             "port-version": {
+              "description": "Deprecated in favor of being a part of 'version' in overrides. Specifies the port-version portion of the version. Overrides the value in one of the other version fields.",
```
Comment (BillyONeal) [inline docs/vcpkg-schema-definitions.schema.json:786]: I just realized I didn't implement this part.

File: src/vcpkg/dependencies.cpp [hunk 3]
```diff
@@ -1729,12 +1729,12 @@ namespace vcpkg
...
-            doc.append_indent().append_raw("\"overrides\": [\n");
-            doc.append_indent(2)
-                .append_raw(fmt::format(R"({{ "name": "{}", "version": "{}" }})", on.name(), baseline.version))
-                .append_raw('\n');
-            doc.append_indent().append_raw("]\n\n");
+            doc.append(msgVersionIncomparable3).append_raw("\n");
+
+            Json::Array example_array;
+            serialize_dependency_override(example_array, DependencyOverride{on.name(), baseline.version});
+            doc.append_raw(Json::stringify_object_member(OVERRIDES, example_array, Json::JsonStyle::with_spaces(2), 1));
```
[the remark this answers is not in the document]
Comment (BillyONeal) [reply src/vcpkg/dependencies.cpp:1736]: I'm not convinced I'm going to do a better job at abstracting this right now. I've raised the level of abstraction to at least json quote strings correctly and I think that's an improvement over the status quo.

File: src/vcpkg/sourceparagraph.cpp [hunk 4]
```diff
@@ -573,7 +587,6 @@ namespace vcpkg
     {
         LocalizedString type_name() const override { return msg::format(msgADependencyFeature); }
 
-        constexpr static StringLiteral NAME = "name";
         constexpr static StringLiteral PLATFORM = "platform";
```
[the remark this answers is not in the document]
Comment (BillyONeal) [reply src/vcpkg/sourceparagraph.cpp:590]: In the interest of making this PR be minimal I would rather do that in a separate PR. I filed https://github.com/microsoft/vcpkg-tool/pull/1355

**Review by ras0219-msft — APPROVED** [review]

**Pull request merged**